timestamp-client/src/main/java/com/palantir/timestamp/RateLimitedTimestampService.java

```diff
@@ -56,13 +62,29 @@
     private final AtomicReference</* nullable */ TimestampHolder> currentBatch =
             new AtomicReference<TimestampHolder>();
 
+    // Required for large internal product.
     public RateLimitedTimestampService(TimestampService delegate, long minTimeBetweenRequestsMillis) {
         this.delegate = TimingProxy.newProxyInstance(TimestampService.class, delegate, timer);
+        this.configSupplier = () -> ImmutableTimestampClientConfig.builder().enableTimestampBatching(true).build();
         this.minTimeBetweenRequestsMillis = minTimeBetweenRequestsMillis;
     }
 
+    // Note that calls to configSupplier *MUST* be cheap, because it is called on every timestamp request.
+    public RateLimitedTimestampService(TimestampService delegate, Supplier<TimestampClientConfig> configSupplier) {
```

Comment:
Right, agree that would be cleaner. I was thinking about doing this now because eventually we'll need to have this know about the config (if we do multi-batching), though actually it's probably fine to recreate the timestamp service in such cases. Will change the design